server/src/main/java/org/allaymc/server/plugin/AllayPluginManager.java

```diff
@@ -81,13 +83,24 @@ protected void onLoad(Map<String, PluginDescriptor> descriptors, Map<String, Plu
 
             var loader = loaders.get(s);
             for (var dependency : descriptor.getDependencies()) {
-                if (plugins.containsKey(dependency.name())) continue;
-
-                if (!dependency.optional()) {
-                    log.error(I18n.get().tr(TrKeys.A_PLUGIN_DEPENDENCY_MISSING, descriptor.getName(), dependency.name()));
-                    iterator.remove();
+                var dependencyContainer = plugins.get(dependency.name());
+
+                if (dependencyContainer == null) {
+                    if (!dependency.optional()) {
```

Comment:
Use guard-statement

Reply:
Sorry for the mindless copy and paste, I didn't see it qwq